src/overlays/actors/ovl_En_Demo_heishi/z_en_demo_heishi.c

```diff
@@ -31,37 +39,150 @@ const ActorInit En_Demo_heishi_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE970C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE975C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE97F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE980C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/EnDemoheishi_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/func_80BE9974.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Demo_heishi/EnDemoheishi_Draw.s")
+static u16 sTextIds[] = { 0x1473 };
+
+void EnDemoheishi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDemoheishi* this = THIS;
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 25.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600D640, &D_06003BFC, this->jointTable, this->morphTable, 17);
+    this->actor.colChkInfo.mass = 0xFF;
+    this->actor.targetMode = 6;
+    this->actor.gravity = -3.0f;
+    Collider_InitAndSetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+    EnDemoheishi_SetupIdle(this);
+}
+
+void EnDemoheishi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDemoheishi* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+}
+
+void EnDemoheishi_ChangeAnimation(EnDemoheishi* this, s32 animIndex) {
+    static AnimationHeader* sAnimations[] = {
+        (AnimationHeader*)0x06006C18, (AnimationHeader*)0x06002A84, (AnimationHeader*)0x06003BFC,
+        (AnimationHeader*)0x06003380, (AnimationHeader*)0x06004770,
+    };
+    static u8 sAnimModes[] = { 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0 };
+
+    this->animIndex = animIndex;
+    this->frameCount = Animation_GetLastFrame(sAnimations[animIndex]);
+    Animation_Change(&this->skelAnime, sAnimations[this->animIndex], 1.0f, 0.0f, this->frameCount,
+                     (u8)(s32)sAnimModes[this->animIndex], -10.0f);
+}
+
+void EnDemoheishi_SetHeadRotation(EnDemoheishi* this) {
+    s16 yawTemp = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+    s32 yaw = ABS_ALT(yawTemp);
+
+    this->headRotXTarget = 0;
+    if ((this->actor.xzDistToPlayer < 200.0f) && (yaw < 20000)) {
+        this->headRotXTarget = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+        if (this->headRotXTarget > 10000) {
+            this->headRotXTarget = 10000;
+        } else if (this->headRotXTarget < -10000) {
+            this->headRotXTarget = -10000;
+        }
+    }
+}
+
+void EnDemoheishi_SetupIdle(EnDemoheishi* this) {
+    EnDemoheishi_ChangeAnimation(this, 0);
+    this->actor.textId = sTextIds[this->unk272 = 0];
+    this->unk270 = 0;
+    this->actionFunc = EnDemoheishi_Idle;
+}
+
+void EnDemoheishi_Idle(EnDemoheishi* this, GlobalContext* globalCtx) {
+    s32 absYawDiff;
+    s16 yawDiff;
+
+    this->actor.flags &= 0xF7FFFFFF;
```

Comment:
```suggestion
    this->actor.flags &= ~0x8000000;
```
For readability, being able to see the flag directly.

Reply:
Fixed